src/getPublicKeys.c

```diff
@@ -0,0 +1,284 @@
+#include "state.h"
+#include "securityPolicy.h"
+#include "uiHelpers.h"
+#include "uiScreens.h"
+#include "getPublicKeys.h"
+
+
+static ins_get_keys_context_t* ctx = &(instructionState.getKeysContext);
+
+static inline void CHECK_STAGE(get_keys_stage_t expected)
+{
+	TRACE("Checking stage... current one is %d, expected %d", ctx->stage, expected);
+	VALIDATE(ctx->stage == expected, ERR_INVALID_STATE);
+}
+
+void respondSuccessEmptyMsg()
+{
+	TRACE();
+	io_send_buf(SUCCESS, NULL, 0);
+	ui_displayBusy(); // needs to happen after I/O
+}
+
+static int16_t RESPONSE_READY_MAGIC = 23456;
+
+bool moreKeysFollow()
+{	
+	switch(ctx->stage) {
+		case GET_KEYS_STAGE_SINGLE_KEY:
+			return false;
+
+		case GET_KEYS_STAGE_GET_KEYS:
+			return ctx->remainingKeys > 0;
+		
+		default:
+			ASSERT(false);
+	}
+	ASSERT(false);
+	return false;
```

Comment:
if the switch covers all cases and even asserts on default case, this is imho more confusing than beneficial as assurance, or was this pattern already adopted in other parts of the codebase?